### PR TITLE
Optimize program enrollments API

### PR DESCRIPTION
**Pull request by jkachel**
# What are the relevant tickets?

#1700 

# Description (What does it do?)

Refactors some queries in the `ProgramSerializer` and `UserProgramEnrollmentDetailsSerializer` and `UserProgramEnrollmentsViewSet` so they should be more efficient. This also removes the implicit program enrollments from the view set - if you're not explicitly enrolled in a program, you won't see it anymore (we should have explicit enrollments currently, though). 

There are more things that could be optimized; I believe the amount of data getting pulled in through CourseSerializer is somewhat excessive too but this should result in a bit of an improvement for now. 

# How can this be tested?

As a user with enrollments in programs, load `/api/program_enrollments`. It should return reasonable data and should do so faster than the same call on the `main` branch. For best results, test each a few times sequentially. 

Additionally, the Dashboard should work as expected. Programs should appear normally; their courses should appear normally; and certificates should display as usual.

### CODE REVIEW

**Comment by pdpinch** [comment]
I don't want to slow this down, but how/can we monitor performance after the new program code is released?

File: courses/serializers.py [hunk 1]
```diff
@@ -390,6 +388,56 @@ class Meta:
         ]
 
 
+class SimpleProgramSerializer(serializers.ModelSerializer):
```
Comment (collinpreston) [inline courses/serializers.py:391]: Does it make sense to update the existing ProgramSerializer?

**Review by collinpreston — COMMENTED** [review]
Looks good, just one comment I'm curious about.

File: courses/serializers.py (hunk 1 above)
Comment (jkachel) [reply courses/serializers.py]: https://github.com/mitodl/mitxonline/pull/1708/commits/5e18fc739fb27ba420f76e75160854d56af5591b does that; there were other fields that the current frontend doesn't use that got pulled into `FullProgramSerializer` that inherits from `ProgramSerializer`.

**Review by collinpreston — APPROVED** [review]
Nice, looks good and is speedy quick!

**Pull request merged**